baseclasses/utils/utils.py

```diff
@@ -20,14 +19,28 @@ def getPy3SafeString(string):
     return string
 
 
-def pp(obj, comm=None):
+def pp(obj, comm=None, flush=True):
+    """
+    Parallel safe printing routine. This method prints ``obj`` (via pprint) on the root proc of ``self.comm`` if it exists. Otherwise it will just print ``obj``.
+
+    Parameters
+    ----------
+    obj : object
+        Any Python object to be printed
+    comm : MPI comm
+        The MPI comm object on this processor
+    flush : bool
+        If True, the stream will be flushed.
+    """
     if (comm is None) or (comm is not None and comm.rank == 0):
         # use normal print for string so there's no quotes
         if isinstance(obj, str):
-            print(obj)
+            print(obj, flush=flush)
         # use pprint for other built-in types (other than string)
-        elif obj.__class__.__module__ == "__builtin__" or isinstance(obj, (CaseInsensitiveDict, CaseInsensitiveSet)):
-            pprint(obj)
+        elif obj.__class__.__module__ == "__builtin__":
```

Comment:
Printing a regular dictionary isn't working for me. Can you check if this is supposed to be `builtins` instead of `__builtin__`?

Reply:
Previously we only used `print` for strings and `pprint` for everything else. Was that causing issues with other classes?

Reply:
Hmm, I can't remember why I changed it. I'll change it back I guess